src/main/java/groovy/lang/GString.java

```diff
@@ -276,4 +309,16 @@ public Pattern negate() {
     public byte[] getBytes(String charset) throws UnsupportedEncodingException {
         return toString().getBytes(charset);
     }
+
+    private boolean checkValuesImmutable() {
+        for (Object value : values) {
+            if (null == value) continue;
+            if (!(IMMUTABLE_TYPE_LIST.contains(value.getClass())
```

Comment:
Perhaps we could use ImmutablePropertyUtils#isBuiltinImmutable? We'd have to account for primitives. Also check for enums?

Reply:
enum's `toString` may return different string literal for each time they are called, e.g.
```groovy
enum Color {
    RED, GREEN, BLUE
    
    String toString() {
        return "${new Date()}"
    }
}
println Color.RED
```

Reply:
Ah yes, I hadn't thought about that case.

Reply:
Tweaked according to your suggestion:
https://github.com/apache/groovy/pull/1309/commits/6187f3e4355db563a3d39d4698ec1b2b064b3406